community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongResourceCollections.java

```diff
@@ -0,0 +1,139 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.collection.primitive;
+
+import java.util.Arrays;
+
+import org.neo4j.graphdb.Resource;
+
+import static org.neo4j.collection.primitive.PrimitiveLongCollections.resourceIterator;
+
+public class PrimitiveLongResourceCollections
+{
+    public static PrimitiveLongResourceIterator iterator( Resource resource, final long... items )
+    {
+        return resourceIterator( PrimitiveLongCollections.iterator( items ), resource );
+    }
+
+    private static final PrimitiveLongResourceIterator EMPTY = new PrimitiveLongBaseResourceIterator( null )
+    {
+        @Override
+        protected boolean fetchNext()
+        {
+            return false;
+        }
+    };
+
+    public static PrimitiveLongResourceIterator emptyIterator()
+    {
+        return EMPTY;
+    }
+
+    public static PrimitiveLongResourceIterator concat( PrimitiveLongResourceIterator... primitiveLongResourceIterators )
+    {
+        return concat( Arrays.asList( primitiveLongResourceIterators ) );
+    }
+
+    public static PrimitiveLongResourceIterator concat( Iterable<PrimitiveLongResourceIterator> primitiveLongResourceIterators )
+    {
+        return new PrimitiveLongConcatingResourceIterator( primitiveLongResourceIterators );
+    }
+
+    public abstract static class PrimitiveLongBaseResourceIterator extends PrimitiveLongCollections.PrimitiveLongBaseIterator
+            implements PrimitiveLongResourceIterator
+    {
+        private Resource resource;
+
+        PrimitiveLongBaseResourceIterator( Resource resource )
+        {
+            this.resource = resource;
+        }
+
+        @Override
+        public void close()
+        {
+            if ( resource != null )
+            {
+                resource.close();
+                resource = null;
+            }
+        }
+    }
+
+    public static class PrimitiveLongConcatingResourceIterator extends PrimitiveLongCollections.PrimitiveLongConcatingIterator
+            implements PrimitiveLongResourceIterator
+    {
+        private final Iterable<PrimitiveLongResourceIterator> iterators;
+        private volatile boolean closed;
+
+        PrimitiveLongConcatingResourceIterator( Iterable<PrimitiveLongResourceIterator> iterators )
+        {
+            super( iterators.iterator() );
+            this.iterators = iterators;
+        }
+
+        @Override
+        protected boolean fetchNext()
+        {
+            return !closed && super.fetchNext();
+        }
+
+        @Override
+        public void close()
+        {
+            if ( !closed )
+            {
+                closed = true;
+                closeAll();
+            }
+        }
+
+        // Copied from IOUtils
```

Comment:
Any specific reason that we copy this instead of call it? :)

Reply:
Introducing a dependency on `neo4j-io` would create a circular dependency.